### ISSUE TITLE
E-mail edit/preview UnicodeDecodeError (client)

### ISSUE THREAD

**Issue by Valraven** [securestate/king-phisher#274]
## Issue Description
When adding my mail template and wanting to edit or preview it, it gives a unicode decode error. My guess is that it has to do with the e-mail being in Danish. But I am not sure.

## Reproduction Steps
1. Start King Phisher
2. Create campaign, all goes well
3. Add e-mail template
4. Open edit or preview and there we go

## Environment Details
Server OS: Kali Linux latest build
King Phisher Version: 1.10
Client OS: Windows 10, fully updated
King Phisher version: 1.10

## Error Details / Stack Trace
```Error Type: builtins.UnicodeDecodeError
Error Details: UnicodeDecodeError('utf-8', b'<!DOCTYPE html>\r\n<html>\r\n<meta http-equiv="Content-Type" content="text/html; charset=utf-8">\r\n<table style="margin-top: 0px; margin-bottom: 40px;" border="0" width="600" align="center">\r\n<tbody>\r\n<tr>\r\n<td>\r\n<div>&nbsp;</div>\r\n</td>\r\n</tr>\r\n\r\n<tr>\r\n<td style="font-weight: normal; font-family: Arial,sans-serif; padding-top: 0px; padding-bottom: 30px; font-size: 36px; line-height: 145%;">\r\n<em>K\xe6re&nbsp;{{first_name}}</em>\r\n</tr>\r\n<tr>\r\n\t<td style="font-weight: normal; font-family: Arial,sans-serif; padding-bottom: 30px; font-size: 18px; line-height: 145%;">\r\n\t\t\tHar du Danmarks bedste kollega, eller m\xe5ske verdens bedste?<br><br>\r\n\t\t\tNominering.eu er p\xe5 udkig efter Danmarks bedste kollega, og udlodder derfor, i samarbejde med <b>Den Europ\xe6riske Union</b>, hele TO iPhone X 64GB. En til dig, og en til din nominerede kollega.\r\n\t\t\t<br><br>\r\n\t\t\tFor at deltage i konkurrencen skal du blot f\xf8lge nedenst\xe5ende link og udfylde formularen.\r\n\t\t\t<br><br>\r\n\t\t\t<b>HUSK: Vi vil meget gerne vide lige pr\xe6cis hvorfor, din kollega er Danmarks bedste!</b>\t\t\t\r\n\t\t\t<div style="     \r\n\t\t\twidth: 600; \r\n\t\t\theight: 500px;\t\t\t\r\n\t\t\tmargin-top: 10px;">\r\n\t\t\t\t<img src="https://i.imgur.com/BNen2KC.jpg">\r\n\t\t\t</div>\r\n\t</td>\r\n</tr>\r\n\r\n<tr>\r\n<td style="background-color: rgb(45, 172, 227); cursor: pointer; font-family: Arial,sans-serif; font-size: 25px; font-weight: 600; text-align: center; padding: 15px;">\r\n<a style="color: #ffffff; text-decoration: none;" href="{{url.webserver}}">\r\nVind iPhone X 64GB\r\n</a></td>\r\n</tr>\r\n<tr>\r\n<td style="text-align: center; font-weight: normal; font-family: Arial,sans-serif; padding-top: 20px; padding-bottom: 20px; font-size: 12.6px; line-height: 145%; border-top: solid 1px #cccccc; color: #757575; border-bottom: solid 1px #cccccc;">\r\n<em>Konkurrencen er udarbejdet i samarbejde med Den Europ\xe6riske Union</em></td>\r\n</tr>\r\n</tbody>\r\n</table>\r\n\r\n</html>', 396, 397, 'invalid continuation byte')
Error UID: 061bc781-ad7c-4a7e-8fba-cced7a79a6a3
RPC Error: N/A (Not a remote RPC error)
King Phisher Version: 1.10.0
Platform Version: Windows: 10 10.0.16299  Multiprocessor Free (Frozen=True)
Python Version: 3.4.4
Gtk Version: 3.14.15
Timezone: Europe/Paris

Thread Information:
    Thread-1 (alive=True daemon=False)
    Thread-7 (alive=True daemon=True)
    Thread-6 (alive=True daemon=True)
 => MainThread (alive=True daemon=False)
    Thread-2 (alive=True daemon=True)

Stack Trace:
Traceback (most recent call last):
  File "C:\king-phisher\king_phisher\client\tabs\mail.py", line 1167, in signal_notebook_switch_page
  File "C:\king-phisher\king_phisher\client\tabs\mail.py", line 789, in show_tab
  File "C:\king-phisher\king_phisher\client\tabs\mail.py", line 638, in load_html_file
  File "C:\Python34\lib\codecs.py", line 694, in read
  File "C:\Python34\lib\codecs.py", line 497, in read
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe6 in position 396: invalid continuation byte```

**Comment by zeroSteiner** [comment]
Check the [File Encoding](https://github.com/securestate/king-phisher/wiki/Templates#file-encoding) section in the wiki. Basically what you're going to want to do is open the template file in another editor and re-save it into UTF-8. That should fix the issue and then you can reload it into King Phisher.

It's the same thing for web pages as well, if you run into it again server side.

**Comment by Valraven** [comment]
That fixed it, at least when it comes to the edit pane and the preview pane. But for some reason the client crashes without giving me an error on why :S

**Comment by zeroSteiner** [comment]
When you say that the client crashes, what exactly are you experiencing? Is it just closing, do you see another exception dialog or some other error message?

**Comment by Valraven** [comment]
It is just closing and I get a windows pop-up that tells me that the program closed unexpectedly, no other errors than that.
<img width="291" alt="error" src="https://user-images.githubusercontent.com/23186396/37772262-5a37c872-2dda-11e8-9bc0-de59c37590d3.png">

Translation: King Phisher Client has stopped functioning.

A problem as arrised, that kept the program from running correctly. Windows has closed the program and gives you a message, when a solution has been found. (like that ever happens lol)

**Comment by zeroSteiner** [comment]
What exactly were you doing when you experienced that error?

**Comment by Valraven** [comment]
I opened the preview tab for the e-mail. Nothing more, nothing less.

**Comment by zeroSteiner** [comment]
Well we can take a look at the issue if you want to share the email template with us. Otherwise there's not a whole lot we can do. You can put it on a public paste site or whatever works for you.

**Comment by Valraven** [comment]
Is it possible to e-mail it? although I believe it is also in the error report at the beginning.

**Comment by zeroSteiner** [comment]
The one at the beginning should have been the one before you modified the encoding. If that's the same file then you need to make sure that you re-encode it correctly into UTF-8. We would need the exact one that's causing the crash. And no, I'd prefer to not get it via email. You could join our [Slack](https://king-phisher-slackin.herokuapp.com/) and send it to me that way if you want it relatively private.

**Comment by wolfthefallen** [comment]
Hard crash error resolved in commit bfc4307b15f9508094d00671a63797881df987af
Handling for non unicode files resolved in commit d4ba2816423e9512364001e406f9512455b85430
